spot_driver/spot_driver/spot_ros2.py

```diff
@@ -844,6 +844,31 @@ def __init__(self, parameter_list: Optional[typing.List[Parameter]] = None, **kw
                 if self.auto_stand.value:
                     self.spot_wrapper.stand()
 
+        # Explicitly get a lease when get_lease_on_action is False
+        # because spot_wrapper does not get a lease at initialization.
+        if self.spot_wrapper is not None and not self.get_lease_on_action.value:
+            self.spot_wrapper.getLease()
+        self.create_service(
+            srv_type=Trigger,
+            srv_name="gen_new_spot_ros_lease",
+            callback=self.new_lease_callback,
+            callback_group=self.group,
+        )
+
+    def new_lease_callback(self, request: Trigger.Request, response: Trigger.Response) -> Trigger.Response:
+        self.get_logger().info("Incoming request to generate a new lease.")
+        if self.spot_wrapper is None:
+            response.success = True
+            response.message = "SpotROS is running in mock mode."
```

Comment:
nit: Use either `spot_ros2` (the node name) or `spot_driver` (the package name)

Reply:
Sounds good, changed to `spot_ros2`.